src/gluonts/mx/trainer/_base.py

```diff
@@ -198,6 +206,13 @@ def base_path() -> str:
                 static_shape=True,
             ):
                 batch_size = train_iter.batch_size
+                sample_batch = next(iter(train_iter))
```

Comment:
Could it be that consuming a batch here leaves the data loader modified just enough to change some results in tests? The tests are definitely to blame, but maybe this surfaces an undesired property of the data loader?

Reply:
yeah, maybe. 

It could be a problem with the dataloader or perhaps the tests are so brittle that small changes (e.g. shuffling the lines in the files) breaks things. I'll try to debug this.

Reply:
In the data loader some internal iterator is constructed at construction time, when that should probably happen at `iter` time, I'll also look into that

Reply:
Yeah, I think this is actually intended behaviour, since we want the next epoch to continue where the previous ended. The failing test is still strange though. I can't reproduce it on my laptop or a docker image. 

For now I changed things to not pull an extra batch for this.